src/Cognito.re

```diff
@@ -83,4 +165,50 @@ let signUp =
         });
       };
     });
+};
+
+open SignInHelper;
+external makeSignInResponse: 't => signInResponseDecoder = "%identity";
+external toJsDict: 'a => Js.Dict.t(Js.Json.t) = "%identity";
+let signIn = (~username: string, ~password: string, ()) => {
+  let signInParams = authenticationParameters(~username, ~password)->toJsDict;
+  /* create and object of this shape using our `authenticationParameters` type
+     ```json
+     {
+       AuthParameters : {
+         "USERNAME": "idkjs",
+         "PASSWORD": "somepassword",
+       }
+     }
+     ```
+     At this point we know we have a valid js object which can test with `Js.Json.test(signInParams, Object)`
```

Comment:
I don't think you logic here is wrong, btw, I just prefer to lean on the type system as much as possible and remove all doubt. Here I think we're talking about 4-5 lines of boilerplate and we're golden.